### ISSUE TITLE
Inquiry about the rapidflow training?

### ISSUE THREAD

**Issue by cppppp** [hmorimitsu/ptlflow#83]
Hi,  I'm trying to train the rapidflow model, but find that the configs are different from the paper. In ptlflow/models/rapidflow/configs/rapidflow-train1-chairs.yaml, it is set that 'max_epochs: 45', 'train_batch_size: 1'. I have revised max_epochs to 10 and train_batch_size to 8 according to the paper, but the result validated on kitti and sintel training set is still worse than the paper. Should I revise other parameters? Thank you very much!

**Comment by hmorimitsu** [comment]
Hi, thanks for the message.

Yes, the configs for chairs and things were wrong. Thanks for pointing this out. I have committed a fix now.

You shouldn't have to change anything else. However, notice that the results on the paper were collected after stage 2 training (Chairs and then Things3D). So, if you have just trained with the FlyingChairs dataset, the results will be worse than in the paper.

If you have any more questions, just let me know.

Best.

**Comment by cppppp** [comment]
Thanks for changing the configs for chairs and things! However, after training with the updated configurations, I am still seeing worse results than the reported ones. Since I’m new to this field, it’s possible that I’ve made some mistakes during training. When training only with the chairs dataset, I get the following validation results:
sintel-clean-val/epe,2.6188
sintel-final-val/epe,3.6041
kitti-2015-val/epe,12.3858
In contrast, when validating with the model provided at https://github.com/hmorimitsu/ptlflow/releases/download/weights1/rapidflow-chairs-9c8c182a.ckpt, the results are better:
sintel-clean-val/epe,2.5049
sintel-final-val/epe,3.6802
kitti-2015-val/epe,10.4116

Similarly, in the second stage of training with the things dataset, my results are still worse than those reported. Could you please provide any suggestions on where I might be going wrong?

I appreciate your help!

Best.

**Comment by hmorimitsu** [comment]
Hello. Judging from these results, I think there is no problem with your training. There is always some oscillation on the validation metrics during the training, and I feel the difference you are seeing is caused by the randomness of the training. If you want to try to improve these results, you can try to use a different seed, or add a few more iterations to the training.

As for the Things3D stage, it may also have been caused by randomness, but it depends on how far your results are. One important thing is to check whether you downloaded the `RGB images (finalpass)` of the FlyingThings3D dataset, as it doubles the amount of training samples. If the dataset is setup correctly, you should see these messages:
```bash
2025-03-06 15:15:10.435 | INFO     | ptlflow.data.datasets:_log_status:256 - Loading 96336 samples from FlyingThings3D dataset.
2025-03-06 15:15:10.530 | INFO     | ptlflow.models.base_model.base_model:configure_optimizers:515 - --trainer.max_steps is not provided. max_steps will be set as 240840 (10 epochs * 24084 steps per epoch / 1 devices / 1 accumulate grad steps)
```

I hope that helps.

Best.

**Comment by cppppp** [comment]
Thanks for your suggestion. I will have a try.

**Comment by cppppp** [comment]
Sorry for bothering again, but I still fail to reach the ideal result. I have tried with different epochs and random seeds, but the best result I could get when training with flyingthings3D is:
sintel-clean-val/epe,1.73347869743016
sintel-final-val/epe,3.0343133672628118
kitti-2015-val/epe,6.888119718283415
kitti-2015-val/flall,20.257289679050444
I have conformed that the sample number is 96336, and the pretrained model is loaded with that provided at https://github.com/hmorimitsu/ptlflow/releases/download/weights1/rapidflow-chairs-9c8c182a.ckpt. Could you please check whether there is anything wrong with the code?

**Comment by hmorimitsu** [comment]
Hi, thanks for the update. Ok, I'll check the model here and see if something has changed.

**Comment by cppppp** [comment]
Thank you so much for taking the time to double-check the code again. Please feel free to tackle it whenever it's most convenient for you.

**Comment by hmorimitsu** [comment]
Hi. The training problem (hopefully) has been fixed at [72b7579](https://github.com/hmorimitsu/ptlflow/pull/90/commits/72b7579d6347c81dec9bcf1545c2f2f9156a38cf). There was an extra bias in the middle of the two separable convolutions.

Thank you again for finding the error. If you happen to have any other problems, please feel free to report them again.

Here is the result of my training after the fix:

![Image](https://github.com/user-attachments/assets/153c216e-3c14-453f-b2c3-80e17397e4c6)

**Comment by cppppp** [comment]
Thank you so much. The result is correct now!

**Comment by cppppp** [comment]
Hi. Although the training with things is confirmed to be correct now, there seems to be something with chairs. I trained with chairs two months ago, and it generated similar results as reported. However, training with the new code leads to undesirable results:
sintel-clean-val/epe,4.294294502346363
sintel-final-val/epe,5.246970780129391
kitti-2015-val/epe,21.72489173412323
kitti-2015-val/flall,51.1386790406704
I'm now trying to figure out the cause. It’s possible that the problem is due to an error on my part.

![Image](https://github.com/user-attachments/assets/6986d44c-af7e-4114-b059-c3854554ce44)

**Comment by cppppp** [comment]
According to my experiment, when training with chairs, the bias of Next1d should still be set as:
```python
x = F.conv2d(
      input,
      self.weight_h,
      self.bias,
      self.stride,
      (0, self.padding),
      self.dilation,
      self.groups,
  )
  x = F.conv2d(
      x,
      self.weight_v,
      self.bias,
      self.stride,
      (self.padding, 0),
      self.dilation,
      self.groups,
  )
```
Then modify the bias in the first conv2d to None, and train with things. This generates similar results as reported. Thank again for your great contribution!
